OpenRA.Game/Actor.cs

```diff
@@ -366,6 +368,11 @@ public bool IsTargetableBy(Actor byActor)
 			return false;
 		}
 
+		public ITargetablePositions[] GetEnabledTargetablePositionTraits()
+		{
+			return TargetablePositionTraits.Where(Exts.IsTraitEnabled).ToArray();
+		}
+
```

Comment:
I would probably ditch this method - it's not doing anything the caller can't do.

Reply:
Also food for thought - the `ToArray` here might not be worthwhile - `Any` (inside the `Positions` property) isn't likely to enumerate the whole sequence. If I'm wrong and avoiding two enumerations is important - we're probably better dropping to a raw loop and avoiding LINQ.